chain/indexer/integrated/processor/state_test.go

```diff
@@ -0,0 +1,357 @@
+	require.Equal(t, chaineconomics.NewTask(nil), proc.TipsetProcessors[tasktype.ChainEconomics])
+	require.Equal(t, consensus.NewTask(nil), proc.TipsetProcessors[tasktype.ChainConsensus])
+}
+
+func TestMakeProcessorsTipSets(t *testing.T) {
+	tasks := []string{
+		tasktype.Message,
+		tasktype.GasOutputs,
+		tasktype.BlockMessage,
+		tasktype.ParsedMessage,
+		tasktype.Receipt,
+		tasktype.InternalMessage,
+		tasktype.InternalParsedMessage,
+		tasktype.MessageGasEconomy,
+		tasktype.MultisigApproval,
+	}
+	proc, err := processor.MakeProcessors(nil, tasks)
+	require.NoError(t, err)
+	require.Len(t, proc.TipsetsProcessors, len(tasks))
+
+	require.Equal(t, message.NewTask(nil), proc.TipsetsProcessors[tasktype.Message])
+	require.Equal(t, gasoutput.NewTask(nil), proc.TipsetsProcessors[tasktype.GasOutputs])
+	require.Equal(t, blockmessage.NewTask(nil), proc.TipsetsProcessors[tasktype.BlockMessage])
+	require.Equal(t, parsedmessage.NewTask(nil), proc.TipsetsProcessors[tasktype.ParsedMessage])
+	require.Equal(t, receipt.NewTask(nil), proc.TipsetsProcessors[tasktype.Receipt])
+	require.Equal(t, internalmessage.NewTask(nil), proc.TipsetsProcessors[tasktype.InternalMessage])
+	require.Equal(t, internalparsedmessage.NewTask(nil), proc.TipsetsProcessors[tasktype.InternalParsedMessage])
+	require.Equal(t, gaseconomy.NewTask(nil), proc.TipsetsProcessors[tasktype.MessageGasEconomy])
+	require.Equal(t, msapprovals.NewTask(nil), proc.TipsetsProcessors[tasktype.MultisigApproval])
+}
+
+func TestMakeProcessorsReport(t *testing.T) {
+	proc, err := processor.MakeProcessors(nil, []string{processor.BuiltinTaskName})
+	require.NoError(t, err)
+	require.Len(t, proc.ReportProcessors, 1)
+	require.Equal(t, indexer.NewTask(nil), proc.ReportProcessors[processor.BuiltinTaskName])
+}
+
+func TestMakeProcessorsInvalidTaskName(t *testing.T) {
+	t.Run("single invalid name", func(t *testing.T) {
+		invalidTask := "invalid_task_name"
+		proc, err := processor.MakeProcessors(nil, []string{invalidTask})
+		require.Error(t, err)
+		require.Nil(t, proc)
+	})
+
+	t.Run("mix of invalid and valid", func(t *testing.T) {
+		validName := tasktype.Message
+		invalidName := "invalid_task_name"
+		proc, err := processor.MakeProcessors(nil, []string{validName, invalidName})
+		require.Error(t, err)
+		require.Nil(t, proc)
+
+		proc, err = processor.MakeProcessors(nil, []string{invalidName, validName})
+		require.Error(t, err)
+		require.Nil(t, proc)
+	})
+}
+
+func TestMakeProcessorsAllTasks(t *testing.T) {
+	// If this test fails it indicates a new processor and/or task name was added and test should be created for it in one of the above test cases.
+	proc, err := processor.MakeProcessors(nil, append(tasktype.AllTableTasks, processor.BuiltinTaskName))
+	require.NoError(t, err)
+	require.Len(t, proc.ActorProcessors, 21)
+	require.Len(t, proc.TipsetProcessors, 5)
+	require.Len(t, proc.TipsetsProcessors, 9)
+	require.Len(t, proc.ReportProcessors, 1)
```

Comment:
serves as a reminder to implementors to update this test file when/if a new processor is added.